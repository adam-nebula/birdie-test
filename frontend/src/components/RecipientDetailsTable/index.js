import React from "react";
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableContainer from '@material-ui/core/TableContainer';
import Paper from '@material-ui/core/Paper';
import RecipientTableRow from "../RecipientTableRow";
import { sortByTimestamp } from "../../services/dataProcessing";
import RecipientTableHeader from "../RecipientTableHeader";

export default function RecipientDetailsTable({ tableSchema, events }) {

    if (tableSchema.length < 1 || events.length < 1) {
        return <p style={{fontSize: "24px"}}>No events to observe for the currently selected event type</p>
    }

    return (
        <TableContainer component={Paper}>
            <Table>
                <RecipientTableHeader tableSchema={tableSchema}/>
                <TableBody>
                    {events.sort((a, b) => sortByTimestamp(a, b)).map(event => (
                        <RecipientTableRow event={event} tableSchema={tableSchema} key={"row-" + event.id}/>
                    ))}
                </TableBody>
            </Table>
        </TableContainer>
    )
}