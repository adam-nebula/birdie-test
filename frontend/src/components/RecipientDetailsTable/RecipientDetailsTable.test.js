import React from "react";
import renderer from 'react-test-renderer';
import RecipientDetailsTable from ".";

it("recipient table renders when has events", () => {

    const event1 = {id: "some_guid_that_shouldn't_display", 
    timestamp: "2019-05-01T07:26:06.684Z",
    event_type: "This_also_shouldn't_display",
    note: "some information",
    };

    const event2 = {id: "some_other_guid_that_shouldn't_display", 
    timestamp: "2019-05-02T07:26:06.684Z",
    event_type: "This_also_shouldn't_display",
    note: "some information again",
    };

    const events = [event1, event2];

    const tableSchema = [{value: "note", label: "Note"}, {value: "timestamp", label: "Timestamp"}];

    const tree = renderer.create(
        <RecipientDetailsTable
        events={events}
        tableSchema={tableSchema}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
})