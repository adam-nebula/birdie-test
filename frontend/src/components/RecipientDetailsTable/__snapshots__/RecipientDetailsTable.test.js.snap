// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recipient table renders when has events 1`] = `
<div
  className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <table
    className="MuiTable-root"
    role={null}
  >
    <thead
      className="MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          Note
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          Timestamp
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          some information again
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          May 2nd 2019, 8:26:06 am
        </td>
      </tr>
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          some information
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          May 1st 2019, 8:26:06 am
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
